src/ui/pages/zcl_abapgit_gui_page_sett_remo.clas.abap

```diff
@@ -984,19 +1017,35 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_SETT_REMO IMPLEMENTATION.
 
     ENDCASE.
 
+    IF mo_popup_picklist IS BOUND. " Uniform popup state handling
+      " This should happen only for a new popup because
+      " on the first re-render main component event handling is blocked
+      " and not called again until the popup distruction
+      IF mo_popup_picklist->is_in_page( ) = abap_true.
+        rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render.
+      ELSE.
+        rs_handled-state = zcl_abapgit_gui=>c_event_state-new_page.
+        rs_handled-page  = zcl_abapgit_gui_page_hoc=>create(
+          ii_child_component = mo_popup_picklist
+          iv_show_as_modal   = abap_true ).
+      ENDIF.
+    ENDIF.
+
+
     " If staying on form, initialize it with current settings
-    IF rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render.
-      mo_form = get_form_schema( is_settings  = ms_settings_old
-                                 io_form_data = mo_form_data ).
-      CREATE OBJECT mo_form_util
-        EXPORTING
-          io_form = mo_form.
+    IF rs_handled-state = zcl_abapgit_gui=>c_event_state-re_render AND mo_popup_picklist IS NOT BOUND.
+      " TODO refactor: same as in constructor
```

Comment:
I'm not quite sure what this code (in this IF block) is doing ... looks like a part of the construction.

Reply:
That block came from https://github.com/abapGit/abapGit/pull/5483

Usually, the schema is defined only in the constructor and never changed. But here, after form values change - like switching online/offline, or changing the head type -, the form schema needs to be updated.

Reply:
Aha. Thanks. This explains indeed. I started some refactoring to fix it. But it appears to be larger than I expected. So in a separate pr right after this one. 

Maybe will experiment with layout definition too. It's full of IFs. I think it can be improved.